lib/Dredd.js

```diff
@@ -27,17 +27,19 @@ function prefixErrors(decoratedCallback, prefix) {
 
 
 function readLocations(locations, options, callback) {
-  if (typeof options === 'function') { [options, callback] = [{}, options]; }
+  const usesOptionsFunction = typeof options === 'function';
```

Comment:
Perhaps this could be called `usesOptions` - a boolean saying whether the caller provided optional `options` argument or not:

```js
const usesOptions = typeof options !== 'function';
```